contracts/Vault.sol

```diff
@@ -148,18 +147,30 @@ contract Vault is ExchangeRate, OwnableUpgradeSafe, PausableUpgradeSafe {
         require(aTokenAmount >= depositorToPrincipal[msg.sender], "atoken must be more than depositorPrincipal");
         require(depositorReserve >= aTokenAmount.sub(depositorToPrincipal[msg.sender]), "atoken must be more than depositorPrincipal");
         depositorReserve = depositorReserve.sub(
-            aTokenAmount.sub(depositorToPrincipal[msg.sender]) 
+            aTokenAmount.sub(depositorToPrincipal[msg.sender])
         );
         require(principal >= depositorToPrincipal[msg.sender], "principal must be over depositorPrincipal");
         principal = principal.sub(depositorToPrincipal[msg.sender]);
         depositorToPrincipal[msg.sender] = 0;
         // withdraw atokens
         aToken.approve(address(aavePool), aTokenAmount);
-        uint256 withdrawn = aavePool.withdraw(
-            address(token),
-            aTokenAmount,
-            msg.sender
-        );
+        uint256 aTokenBalance = aToken.balanceOf(address(this));
+
+        // @dev in an edge case where the btokenToToken() math is imprecise,
+        // @dev aTokenAmount could be higher than total aToken balance when the withdraw is owned by single owner
```

Comment:
if you figure out how to get the vault into this state, please add a description of how it gets into this state. is the state correct?

i'm not clear that this fixes the real bug. it just fixes a negative effect of the real bug?

i guess we need to ask how can aTokenAmount ever be higher than aTokenBalance?

should the bug be fixed by fixing how the withdrawal works - is that really the bug? -- or should the situation where  "aTokenAmount could be higher than total aToken balance when the withdraw is owned by single owner" never be possible in any situation. with multiple or single owners.

Reply:
it's a little bit of a struggle trying to test this state, I for some reason am unable to replicate it in my local environment. However, what we do know is that the error thrown from AAVE indicates that we are trying to withdraw too much aToken than we have. Plus after running another deployment with this code changes: I no longer see the error.

Shall we think about how to break the state transition function and withdrawals apart, so we can make different units independently testable? Perhaps we can plan it out and execute it in the next set of PRs.